### PR TITLE
Add waypoint following behavior

### PR DESCRIPTION
**Pull request by hidmic**
# Description

This patch revamps the `GoToWaypoint` action and brings back position control into the `autopilot`.

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [x] Unit tests are passing
- [x] Linter tests are passing

# How To Test

A. Run `mission_control` integration tests:

```sh
catkin_make run_tests_mission_control
```

B. Perform manual QA i.e. send vehicle to an actual mission.

### CODE REVIEW

**Comment by ChrisScianna** [comment]
Go for it

File: mission_control/include/mission_control/behaviors/waypoint.h [hunk 1]
```diff
@@ -40,6 +40,7 @@
 #include <behaviortree_cpp_v3/basic_types.h>
 #include <behaviortree_cpp_v3/behavior_tree.h>
 #include <behaviortree_cpp_v3/bt_factory.h>
+#include <geodesy/utm.h>
```
Comment (patapufate) [inline mission_control/include/mission_control/behaviors/waypoint.h:43]: Remove #include "mission_control/behavior.h"

File: mission_control/src/behaviors/waypoint.cpp [hunk 2]
```diff
@@ -34,222 +34,116 @@
...
-  getInput<double>("time_out", timeOut_);
-  subStateData_ =
-      nodeHandle_.subscribe("/pose/corrected_data", 1, &GoToWaypoint::stateDataCallback, this);
-
-  goalHasBeenPublished_ = false;
-  waypointBehaviorPub_ = nodeHandle_.advertise<mission_control::Waypoint>("/mngr/waypoint", 1);
-}
+  if (!getInput<double>("latitude", latitude_) || !getInput<double>("longitude", longitude_))
+  {
+    throw std::runtime_error("Either latitude or longitude were left unspecified");
```
Comment (patapufate) [inline mission_control/src/behaviors/waypoint.cpp:50]: should we throw BT::RuntimeError instead?

File: mission_control/test/test_files/test_missions/waypoint_mission_test.xml [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
- </root>
+<root main_tree_to_execute="main" >
+  <BehaviorTree ID="main">
+    <Sequence>
+      <GoToWaypoint
+          latitude="3.0"
+          longitude="4.0"
+          altitude="2.0"
+          speed_knots="6.0"
+          tolerance_radius="0.1"/>
```
Comment (patapufate) [inline mission_control/test/test_files/test_missions/waypoint_mission_test.xml:11]: Should we mantein the time_out port for being tested by test_if_mission_control_aborts_when_behavior_returns_time_out_failure.test?

Comment (hidmic) [reply mission_control/test/test_files/test_missions/waypoint_mission_test.xml:11]: I was hoping we can use Timeout decorators.

File: mission_control/include/mission_control/behaviors/waypoint.h (hunk 1 above)
Comment (hidmic) [reply mission_control/include/mission_control/behaviors/waypoint.h:43]: Good catch. See https://github.com/ChrisScianna/ROS-Underwater-RnD/pull/118/commits/1c6eb422880713214455b79b6a5e3b7fb38f4d8a.

File: mission_control/src/behaviors/waypoint.cpp (hunk 2 above)
Comment (hidmic) [reply mission_control/src/behaviors/waypoint.cpp]: Good idea. See https://github.com/ChrisScianna/ROS-Underwater-RnD/pull/118/commits/1c6eb422880713214455b79b6a5e3b7fb38f4d8a.

**Review by patapufate — APPROVED** [review]
LGTM!

**Comment by hidmic** [comment]
Tests for this behavior alone pass. Moving forward. We'll get around to fix the other ones soon.

**Pull request merged**